templates/repo/issue/view_content/watching.tmpl

```diff
@@ -0,0 +1,13 @@
+<form hx-boost="true" hx-sync="this:replace" hx-target="this" hx-push-url="false" hx-swap="show:no-scroll" method="post" action="{{.Issue.Link}}/watch">
```

Comment:
This is a general question for clarity, but is there any way to reduce the amount of `hx-` attributes we use here? It really makes htmx seem very pervasive, contributors will need to actively learn what all of these do and how they work.

Things like `target`, `swap`, `post` seem fairly easy to infer, but `boost`, `sync`, `push-url` seem unintuitive at first glance. (at least imo)

Reply:
Similarly to `hx-headers`, is there some set of reasonable defaults we can have that are override-able?

The effect of `hx-sync="this:replace"` and `hx-swap="show:no-scroll"`, for example, seem like something that would happen fairly commonly.
As an aside, I don't see `show:no-scroll` on the [reference page](https://htmx.org/attributes/hx-swap/)

Reply:
I explained why each attribute is added as a way to make code review easier, I don't see a way to implement it in a more generic manner. The docs for htmx are pretty good so it shouldn't be hard to grasp.

I do see the defaults are tunable (thanks for the idea) so I will commit a change in the defaults, this will also allow me to remove `show:no-scroll`.

Reply:
> I explained why each attribute is added as a way to make code review easier, I don't see a way to implement it in a more generic manner. The docs for htmx are pretty good so it shouldn't be hard to grasp.

Sorry, to clarify I meant for new contributors. You did a great job summarizing above in the context of this PR. 

I think tuning the defaults would be great, would be interesting to see if a common pattern emerges for what defaults we use most often.

Reply:
The default tuning is here https://github.com/go-gitea/gitea/pull/28908/commits/2096553aba4e817ff0ad80ee1045ef8b5c34cc9d

Reply:
I think another thing we could add to `<body>` might be `hx-push-url`?
https://htmx.org/attributes/hx-push-url/
> `hx-push-url` is inherited and can be placed on a parent element

Since we're (at least for now) mainly looking at replacing components, my assumption would be we nearly never want to change the URL, right?

Reply:
Sounds good, will do.

Reply:
Much nicer https://github.com/go-gitea/gitea/pull/28908/commits/90b2eb3faf516407fd51d1ee130b3027a2eb4c64